ironfish/src/rpc/adapters/tlsAdapter.ts

```diff
@@ -37,6 +42,15 @@ export class RpcTlsAdapter extends RpcSocketAdapter {
   protected async getTlsOptions(): Promise<tls.TlsOptions> {
     const nodeKeyExists = await this.fileSystem.exists(this.nodeKeyPath)
     const nodeCertExists = await this.fileSystem.exists(this.nodeCertPath)
+    const rpcAuthToken = this.node.internal.get('rpcAuthToken')
+
+    if (!rpcAuthToken || rpcAuthToken === '') {
+      this.logger.debug(
+        `Missing RPC Auth token in internal.json config. Automatically generating auth token.`,
+      )
+      this.node.internal.set('rpcAuthToken', uuid())
+      await this.node.internal.save()
+    }
```

Comment:
This is implemented in both the TlsAdapter and below in the SDK; do we need both?

Passing the node into the adapter creates a circular reference

Reply:
I will keep the one in SDK and delete this one in TlsAdapter